src/answers.sql

```diff
@@ -1,15 +1,87 @@
 -- Your answers here:
 -- 1
-
--- 2
-
+SELECT c.name,
+    COUNT(c.name)
+FROM countries c
+    INNER JOIN states s ON c.id = s.country_id
+GROUP BY c.name;
+--2
+SELECT COUNT(*) employees_without_bosses
+FROM employees
+WHERE supervisor_id IS NULL;
 -- 3
-
+SELECT c.name,
+    o.address,
+    COUNT(e.id) as count
+FROM employees e
+    INNER JOIN offices o ON e.office_id = o.id
+    INNER JOIN countries c ON c.id = o.country_id
+GROUP BY c.name,
+    o.address
+ORDER BY count DESC,
+    c.name
+LIMIT 5;
 -- 4
-
+SELECT supervisor_id,
+    COUNT(supervisor_id) as count
+FROM employees
+GROUP BY supervisor_id
+ORDER BY count DESC
+LIMIT 3;
 -- 5
-
+SELECT COUNT(*) list_of_office
+FROM offices o
+    INNER JOIN states s ON o.state_id = s.id
+WHERE s.name = 'Colorado';
 -- 6
-
+SELECT o.name,
+    COUNT(o.name) as count
+FROM offices o
+    INNER JOIN employees e ON e.office_id = o.id
+GROUP BY o.name
+ORDER BY count DESC;
 -- 7
-
+WITH counts AS (
```

Comment:
This query can be significantly simplified. You already have thE CTE, you could have use that table to easily get the one with most and the one with less employees. There is not need of using having.

Reply:
I resolved and refractored it, and it makes sense to avoid having statement.

```sql
WITH counts AS (
    SELECT o.address, COUNT(o.id) AS count
    FROM offices o
        INNER JOIN employees e ON e.office_id = o.id
    GROUP BY o.address
	ORDER BY count
) 
(
    SELECT * FROM counts
	ORDER BY count DESC
	LIMIT 1 
)
UNION ALL
(
	SELECT * FROM counts
	LIMIT 1 
);
```

This solution avoids using having, and only makes 1 inner join. Leading to using less data on the queries after the CTE.